### DISCUSSION TITLE
Help document Gen4 connection

### DISCUSSION THREAD

**Discussion by wills106** [wills106/homeassistant-solax-modbus#42] [Q&A]
Could someone please Document Ideally with pictures how to connect the Gen 4 to a RS485 - USB adaptor and a RS485 - Ethernet adaptor.

**Comment by infradom** [comment]
OK, I will do so tomorrow. I can document 2 solutions as I now have 2 solutions working: 
- one with the USB to serial RS485 converter (generates 'blocking warnings') (cost 2€)
- one with the "waveshare RS485 to Eth (B)" converter (no warnings) (cost 25 €)
I prefer the waveshare solution, but I have only tested it for a couple of hours
Where should I put this documentation ? In the README.md page ?

**Reply by wills106** [reply]
Yes in the README.md is fine. Perhaps under Installation?
I can always format it after.

```
<details>
  <summary>Gen4 RS485</summary>
Content...
</details>
```

It will help other users out as I know some people are unsure with the RS485 bit.

**Comment by infradom** [comment]
I wrote the documentation, and made some pictures. I believe it is better to host this documentation on a separate page of the wiki
I did this on a wiki page of my fork (that should be in sync):
You can find my work for the time being under: 
https://github.com/infradom/homsassistant-solax-modbus/wiki/Gen4-Installation-Notes
Can you copy / merge this work to wherever you would like it to appear in your main line ?

Thanks,

**Reply by wills106** [reply]
Can you do a PR on a Wiki page? Or do I need to create a Wiki on mine and give you editing permission to copy over your content? Never done any Wiki stuff before. Although I can see it being handy for storing other information such as run modes etc.

**Reply by infradom** [reply]
I dont think pull requests can be done on a wiki page.
It is fairly easy to copy the wiki tekst and images from my wiki to yours. If you give me access rights, I can do this.
As the documentation gets more complicated, a wiki is indeed better for documenting some details

**Reply by infradom** [reply]
Gen4 Installation notes is copied/transferred to your wiki
We still need to let the README file refer to some wiki pages

**Reply by wills106** [reply]
I'll update the README.md now. Also shared the wiki on the community.home-assistant.io as I know some are struggling with connection for Gen4. Shame they dropped the built in ethernet connection.

**Comment by infradom** [comment]
I added a warning in the Gen4 modes of operation wiki page: 
Automations should not modify parameters too frequently, as these modifications are writen to EEPROM and EEPROM has a limited number of write cycles. This statement seems to be added also to the latest modbus protocl documentation from Solax.

I think it is wise to add this warning also to the other generations documentation

**Comment by Mouxy** [comment]
Hi

I thought about adding some notes regarding the 1X-AC for the Wiki:

- The 1X-AC default baud rate is 115200 and I tested both the USB and Waveshare and it works fine, but changing it to 19200 makes more sense I think as it ensures it works with future integration updates (as the codes for the USB uses 19200).
Changing it to 19200 is super easy and doesn't require the installer code (picture attached)

(Love the Braud Rate typo)
![IMG_1612](https://user-images.githubusercontent.com/642149/164070389-a8dc2cf1-f423-4308-91aa-29f1c82e6bc2.jpg)

- The user manual mentions pins 3,4,5 as GND, A, B and as you say, only A,B are needed (4,5)

Here is a picture in case you want to add it to the compatible inverters
![IMG_1611](https://user-images.githubusercontent.com/642149/164070903-c92f6118-38d3-40dd-8ade-a23bde82e47b.jpg)

**Reply by wills106** [reply]
Personally I would recommend keeping your baud rate at 115200 if you want to poll your Inverter quicker. As I am not sure if the slower 19200 would allow you to poll faster? ( I don't know the limitations )

I personally like to poll my Gen3 Hybrid over TCP at 2 second Intervals. It can do 1 second.

At the moment you are best using your Waveshare compared to a RS485 - USB Adaptor as you don't get the blocking errors.
We probably need to set the baud rate adjustable in the future!

Thanks for the pic, I'll add it to the Wiki.

**Reply by infradom** [reply]
I just added the picture .. ;-)
